### PR TITLE
Remove Stack service

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Remove Stack Service, as related code:
- model
- stack permissions
- DAOs ( added DB migration script to drop stack tables )

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13459

### CODE REVIEW

**Comment by mkuznyetsov** [comment]
ci-test

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java [hunk 1]
```diff
@@ -37,26 +38,26 @@
   @Inject private TestWorkspaceProvider testWorkspaceProvider;
 
   public TestWorkspace createWorkspaceFromStackWithProject(
```
Comment (mshaposhnik) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java]: Rename method?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java [hunk 2]
```diff
@@ -37,26 +38,26 @@
...
-      NewWorkspace.Stack stack, String workspaceName, String projectName) {
-    return createWorkspaceFromStack(stack, workspaceName, ImmutableList.of(projectName), null);
+      Devfile devfile, String workspaceName, String projectName) {
+    return createWorkspaceFromStack(devfile, workspaceName, ImmutableList.of(projectName), null);
   }
 
   public TestWorkspace createWorkspaceFromStackWithoutProject(
```
Comment (mshaposhnik) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java]: this one too

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java [hunk 3]
```diff
@@ -37,26 +38,26 @@
...
   }
 
   public TestWorkspace createWorkspaceFromStackWithProjects(
-      NewWorkspace.Stack stack, String workspaceName, List<String> projectNames) {
-    return createWorkspaceFromStack(stack, workspaceName, projectNames, null);
+      Devfile devfile, String workspaceName, List<String> projectNames) {
+    return createWorkspaceFromStack(devfile, workspaceName, projectNames, null);
   }
 
   public TestWorkspace createWorkspaceFromStack(
```
Comment (mshaposhnik) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java:55]: same for the rest?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java (hunk 1 above)
Comment (mkuznyetsov) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/CreateWorkspaceHelper.java]: ok

**Comment by sleshchenko** [comment]
Please review every `stack` word that is left and make sure that it should not be removed.
Like [che.predefined.stacks.reload_on_start](https://github.com/eclipse/che/blob/che-13459/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties#L362) configuration property

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
ci-test

File: multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/filters/PublicPermissionsRemoveChecker.java [hunk 4]
```diff
@@ -1,67 +0,0 @@
...
-import org.eclipse.che.multiuser.permission.workspace.server.stack.StackDomain;
-
-/**
- * Recipe and Stack domains remove permissions checker.
- *
- * @author Anton Korneta
- */
-@Singleton
-public class PublicPermissionsRemoveChecker implements RemovePermissionsChecker {
-
```
Comment (skabashnyuk) [inline multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/filters/PublicPermissionsRemoveChecker.java:36]: @sleshchenko @mkuznyetsov are you sure this only stacks specific class?

File: multiuser/sql-schema/src/main/resources/che-schema/7.0.0-RC5.0/0.1__remove_stack_permissions.sql [hunk 5]
```diff
@@ -0,0 +1,13 @@
+--
```
Comment (skabashnyuk) [inline multiuser/sql-schema/src/main/resources/che-schema/7.0.0-RC5.0/0.1__remove_stack_permissions.sql]: version in file path

Comment (skabashnyuk) [reply multiuser/sql-schema/src/main/resources/che-schema/7.0.0-RC5.0/0.1__remove_stack_permissions.sql]: 7.1??

File: wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta8.0-RC5.0/1__remove_stacks.sql [hunk 6]
```diff
@@ -0,0 +1,15 @@
+--
```
Comment (skabashnyuk) [inline wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta8.0-RC5.0/1__remove_stacks.sql]: file location 7.1 ??

File: wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta8.0-RC5.0/1__remove_stacks.sql [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+-- SPDX-License-Identifier: EPL-2.0
+--
+-- Contributors:
+--   Red Hat, Inc. - initial API and implementation
+--
+
+DROP TABLE stack_components;
+DROP TABLE stack_tags;
+DROP TABLE stack;
```
Comment (skabashnyuk) [inline wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta8.0-RC5.0/1__remove_stacks.sql]: new line

**Review by skabashnyuk — COMMENTED** [review]
in general LGTM

**Comment by l0rd** [comment]
Removing label `do-not-merge/hold`. @mkuznyetsov can you rebase please?

**Review by skabashnyuk — APPROVED** [review]

File: multiuser/integration-tests/che-multiuser-mysql-tck/pom.xml [hunk 8]
```diff
@@ -252,11 +252,7 @@
                                             <port>jdbc.port:3306</port>
                                         </ports>
                                         <wait>
-                                            <tcp>
-                                                <ports>
-                                                    <port>3306</port>
```
Comment (sleshchenko) [inline multiuser/integration-tests/che-multiuser-mysql-tck/pom.xml:257]: Is it cleaning up or changes related to Stack API?

Comment (skabashnyuk) [reply multiuser/integration-tests/che-multiuser-mysql-tck/pom.xml:257]: synced with Postgres. Without that, I can't test it on OSX

File: multiuser/sql-schema/src/main/resources/che-schema/7.1.0/0.1__remove_stack_permissions.sql [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+-- available under the terms of the Eclipse Public License 2.0
+-- which is available at https://www.eclipse.org/legal/epl-2.0/
+--
+-- SPDX-License-Identifier: EPL-2.0
+--
+-- Contributors:
+--   Red Hat, Inc. - initial API and implementation
+--
+
+DROP TABLE che_stack_permissions_actions;
```
Comment (sleshchenko) [inline multiuser/sql-schema/src/main/resources/che-schema/7.1.0/0.1__remove_stack_permissions.sql]: Seems the file must be named as `7.1.0/1.1_remove....`

**Review by sleshchenko — CHANGES_REQUESTED** [review]

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by skabashnyuk** [comment]
ci-test

**Pull request merged**